crates/nu-command/src/filesystem/cp.rs

```diff
@@ -372,239 +94,141 @@ impl Command for Cp {
-            Value::string(msg, span)
-        }
-        Err(e) => convert_io_error(e, src, dst, span),
-    }
-}
-
-// This uses `read` and `write` to copy a file. There is another function called `copy_file`
-// which uses `std::fs::copy` instead which is faster but does not provide progress updates for the copy. try to keep the
-// logic in this function in sync with `copy_file`
-fn copy_file_with_progressbar(
-    src: PathBuf,
-    dst: PathBuf,
-    span: Span,
-    ctrlc_status: &Option<Arc<AtomicBool>>,
-) -> Value {
-    let mut bytes_processed: u64 = 0;
-    let mut process_failed: Option<std::io::Error> = None;
 
-    let file_in = match std::fs::File::open(&src) {
-        Ok(file) => file,
-        Err(error) => return convert_io_error(error, src, dst, span),
-    };
-
-    let file_size = match file_in.metadata() {
-        Ok(metadata) => Some(metadata.len()),
-        _ => None,
-    };
-
-    let mut bar = progress_bar::NuProgressBar::new(file_size);
-
-    let file_out = match std::fs::File::create(&dst) {
-        Ok(file) => file,
-        Err(error) => return convert_io_error(error, src, dst, span),
-    };
-    let mut buffer = [0u8; 8192];
-    let mut buf_reader = BufReader::new(file_in);
-    let mut buf_writer = BufWriter::new(file_out);
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let interactive = call.has_flag("interactive");
+        let force = call.has_flag("force");
+        let no_clobber = call.has_flag("no-clobber");
+        let progress = call.has_flag("progress");
+        let recursive = call.has_flag("recursive");
+        let verbose = call.has_flag("verbose");
 
-    loop {
-        if nu_utils::ctrl_c::was_pressed(ctrlc_status) {
-            let err = std::io::Error::new(ErrorKind::Interrupted, "Interrupted");
-            process_failed = Some(err);
-            break;
+        let debug = call.has_flag("debug");
+        let overwrite = if no_clobber {
+            uu_cp::OverwriteMode::NoClobber
+        } else if interactive {
+            if force {
+                uu_cp::OverwriteMode::Interactive(uu_cp::ClobberMode::Force)
+            } else {
+                uu_cp::OverwriteMode::Interactive(uu_cp::ClobberMode::Standard)
+            }
+        } else if force {
+            uu_cp::OverwriteMode::Clobber(uu_cp::ClobberMode::Force)
+        } else {
+            uu_cp::OverwriteMode::Clobber(uu_cp::ClobberMode::Standard)
+        };
+        #[cfg(any(target_os = "linux", target_os = "android", target_os = "macos"))]
+        let reflink_mode = uu_cp::ReflinkMode::Auto;
+        #[cfg(not(any(target_os = "linux", target_os = "android", target_os = "macos")))]
+        let reflink_mode = uu_cp::ReflinkMode::Never;
+        let mut paths = collect_filepath_arguments(call);
+        if paths.is_empty() {
```

Comment:
This check is not quite enough because `cp foo` should give an error, so the length should be either greater than 1 here, or it should be non-empty after popping the target. Whatever you find ends up looking best :)